### PR TITLE
Use dict for parent ref

### PR DESCRIPTION
**Pull request by ulziibay**
Just modifying the node seem to trigger the pyflakes issue: https://gitlab.com/pycqa/flake8/issues/536
Lets not touch AST nodes.
Also, simplified `UnescapedTemplateFileExtensionsVisitor` as the version we shipped do not handle the edge cases. @minusworld lmk if you think its ok

### CODE REVIEW

File: flake8_flask/main.py [hunk 1]
```diff
@@ -51,29 +52,4 @@ def run(self):
                     node.col_offset,
                     message,
                     visitor.__class__.__name__,
-                )
-
-
-if __name__ == "__main__":
```
Comment (nbrahms) [inline flake8_flask/main.py]: Why is this block removed?

File: flake8_flask/unescaped_template_file_extensions.py [hunk 2]
```diff
@@ -55,26 +54,6 @@ def _is_kwarg_escaped(self, keyword) -> bool:
         elif isinstance(fxn, ast.Name):
             return fxn.id in escape_function_names
 
-    def _edge_case_detect_return_content_type_with_text(
```
Comment (nbrahms) [inline flake8_flask/unescaped_template_file_extensions.py]: And this one?

File: flake8_flask/main.py (hunk 1 above)
Comment (ulziibay) [reply flake8_flask/main.py]: It was unused. As described in https://flake8.pycqa.org/en/latest/plugin-development/registering-plugins.html, Flake8 just uses the class registered as an entrypoint, but do not execute the file in any way!

File: flake8_flask/unescaped_template_file_extensions.py (hunk 2 above)
Comment (ulziibay) [reply flake8_flask/unescaped_template_file_extensions.py]: > simplified UnescapedTemplateFileExtensionsVisitor as the version we shipped do not handle the edge cases.

**Pull request merged**